test/StringClient.test.js

```diff
@@ -64,4 +61,35 @@ describe('StringClient', function () {
             return done();
         });
     });
+
+
+    it('should honor requestTimeout when socket has already been established',
+    function (done) {
+        SERVER.get('/foo', function (req, res, next) {
+            res.send('foo');
+            return next();
+        });
+
+        SERVER.get('/fooSlow', function (req, res, next) {
+            setTimeout(function () {
+                res.send('foo');
+                return next();
+            }, 200);
+        });
+
+        // first request should establish keep alive
+        CLIENT.get('/foo', function (err, req, res, data) {
+            assert.ifError(err);
+            assert.strictEqual(data, 'foo');
+            // second request should reuse existing socket
+            CLIENT.get({
+                path: '/fooSlow',
+                requestTimeout: 100
```

Comment:
What happens if the previous connection had a requestTimeout? Does this "trump" that timeout?

Reply:
Each call gets its own req object created, so there's no overlap.